### DISCUSSION TITLE
Close modal and redirect

### DISCUSSION THREAD

**Discussion by howdu** [wire-elements/modal#256] [Q&A]
How can I redirect instead of showing the modal?

I have a table of videos each with an upload button that loads the upload modal.
`wire:click="$emit('openModal', 'videos.upload', {{ json_encode([$row->id]) }})"`

Inside the modal I fetch the youtube playlists for the user to choose where they want the video to be uploaded.
Before doing this I check if the token as expired and would liek to redirect but it tries to render empty page "Undefined array key 1"

```php
class Upload extends ModalComponent
{
    public array $plastlists;

    public Video $video;

    public function mount($id)
    {
        $this->video = Video::findOrFail($id);

        if (Youtube::isAccessTokenExpired()) {
            $this->skipRender();
            return $this->redirect(Youtube::createAuthUrl());
        }

        $this->plastlists = [];
        $plastlists = $this->getPlaylists('contentDetails,id,snippet,status', ['mine' => true]);
        ....
    }
```

**Comment by yabdab** [comment]
Anyone have answer to this?

**Comment by yabdab** [comment]
```php
$this->skipPreviousModal()->closeModalWithEvents([
           redirect()->route('admin-products')
        ]);
```
